### PR TITLE
setup cypress and jest

### PR DESCRIPTION
**Pull request by SanaBob**
- cypress e2e test
- cypress component test
- jest test

Note: swc and ts-node dev-dep are needed for jest tests

### CODE REVIEW

File: cypress/component/app/pageHome.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
```
Comment (Opeyem1a) [inline cypress/component/app/pageHome.cy.tsx:1]: what is this file supposed to be?

File: jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
+/**
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/configuration
+ */
+
+import type {Config} from 'jest';
```
Comment (Opeyem1a) [inline jest.config.ts:6]: strong believer that we should have added this when we needed it and not in advance, but it's already done now so not a huge deal

File: cypress/support/component-index.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (Opeyem1a) [inline cypress/support/component-index.html:1]: what this?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Install dependencies
+        run: yarn
+      - uses: cypress-io/github-action@v6
+        name: Cypress run
+        with:
+          component: true
+          build: npm run build
+          start: npm start
+          browser: chrome
+  jest-run:
```
Comment (Opeyem1a) [inline .github/workflows/test.yml:32]: really don't love adding all this when we have no tests to be running, at this point it just uses our github actions budget without value

same with cypress honestly but slightly less so

Comment (Opeyem1a) [reply .github/workflows/test.yml:32]: to clarify, my point here is that adding the tests to ci is not inherently a part of adding the test framework to the repo, they should be separate tasks imo and the CI part should only be done when valuable to do so

File: cypress/component/app/pageHome.cy.tsx (hunk 1 above)
Comment (SanaBob) [reply cypress/component/app/pageHome.cy.tsx:1]: It's a sample component test for cypress to check the ci but I will be removing it and split the CI part into another PR

File: cypress/support/component-index.html (hunk 3 above)
Comment (SanaBob) [reply cypress/support/component-index.html:1]: Generated cypress file, I thought it was needed, will be removing this

Comment (SanaBob) [reply cypress/support/component-index.html:1]: Apparently, this is a mandatory file for cypress component testing

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**